src/tests/gtests.cpp

```diff
@@ -174,3 +198,61 @@ TEST_F(LeelaTest, MoveOnOccupiedSq) {
     // Find this error in the output
     EXPECT_NE(output.find("illegal move"), std::string::npos);
 }
+
+// Basic TimeControl test
+TEST_F(LeelaTest, TimeControl) {
+    auto maingame = get_gamestate();
+    std::string output;
+    // clear_board to force GTP to make a new UCTSearch.
+    // This will pickup our new cfg_* settings.
+    GTP::execute(maingame, "clear_board");
+
+    // Absolute time 100s = 1.37s per move in opening.
+    // Enough to be visible on display_times
+    GTP::execute(maingame, "kgs-time_settings canadian 0 120 25");
+    testing::internal::CaptureStderr();
+    GTP::execute(maingame, "showboard");
+    output = testing::internal::GetCapturedStderr();
```

Comment:
I'm strongly against using internal APIs though, they are private for a reason.

Reply:
I was following the style @gcp used, so let's see what his opinion is. Looks like you can do is by redirecting stdout/stderr to a stringstream:

https://stackoverflow.com/questions/3803465/how-to-capture-stdout-stderr-with-googletest

I haven't tried it yet.

Reply:
I basically searched on how to do it and most examples were using that internal API.

Reply:
not working yet. See main thread.